server/server.go

```diff
@@ -7,20 +7,25 @@ import (
 	"io"
 	"log"
 	"net/http"
+	"sync"
 )
 
 // Parses the client request and uploads the file.
 func upload(w http.ResponseWriter, r *http.Request) {
 	fmt.Println("File Upload Endpoint Hit")
 
 	// Give a max limit for file upload.
-	r.ParseMultipartForm(10 << 20)
+	err := r.ParseMultipartForm(10 << 20)
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
 
 	// Returns the first file for the given key 'file'.
 	file, handler, err := r.FormFile("file")
```

Comment:
how do users know that the key should be "file"?
Requires documentation.

Also, make "file" a const and then use it. Magic strings should not be used.